linked-lists/tests/linkedList.test.js

```diff
@@ -53,17 +53,32 @@ describe('reverse tests', () => {
     reverseList.append(1);
     reverseList.append(2);
     reverseList.append(3);
-    console.log(reverseList);
+    // console.log(reverseList);
     reverseList.reverse();
     expect(reverseList.head.value).toEqual(3);
   });
 });
 
-describe('serialize tests', () => {
-  test('', () => {
-    let linkedList = new LinkedList();
-    linkedList.append(1);
-    let actual = linkedList.serialize();
-    expect(typeof actual).toBe('string');
+// describe('serialize tests', () => {
+//   test('', () => {
+//     let linkedList = new LinkedList();
+//     linkedList.append(1);
+//     let actual = linkedList.serialize();
+//     expect(typeof actual).toBe('string');
+//   });
+// });
+describe('insert before tests', () => {
+  let linkedList = new LinkedList();
+  linkedList.append(1);
+  linkedList.append(2);
+  test('should insert 3 before 2', () => {
+    linkedList.insertBefore(2, 3);
+    expect(linkedList.head.next.value).toEqual(3);
   });
-});
\ No newline at end of file
+  //QUESTION: I cant figure out why this test wont pass???
```

Comment:
There are 2 reasons this isn't passing:

1. On line 81, the `tobeNull` method needs parentheses: `toBeNull()`
2. The second reason is because in your actual code (not the tests), your code does not check for undefined values. Specifically, line 113 has a value of `undefined`, and then it tries to set it to a variable. Instead, your code should check for an undefined value, and return `null` if it is true.